### ISSUE TITLE
3.0.6: Error: Call to undefined function wiremail()

### ISSUE THREAD

**Issue by cb2004** [ryancramerdesign/ProcessWire#1667]
I am guessing this has something to do with namespace and/or the file compiler. I have a page template, then given a certain page->id I do an include "forms/contact.php".

Used Compiled File?
Yes (and included files)

Cache
Enabled

Cache disabling POST variables
*

Works fine in a 2.7.x environment.

**Comment by adrianbj** [comment]
Is the message reading wiremail() or wireMail()?

I thought incorrect case was now working with the FileCompiler, but might be worth checking.

**Comment by ryancramerdesign** [comment]
I think the issue might be the include "forms/contact.php", that may not be specific enough since without a leading "./" or "/" the include() function can potentially search all php include paths. It might be preventing the compile. The first thing I would try is: 
`````````
include("./forms/contact.php"); 
``````````
Another possibility is the case of wiremail() which should be wireMail().

**Comment by cb2004** [comment]
My variable is assigned with:

$mail = wireMail();

But the error is:

Error: Call to undefined function wiremail()

I will check my include and report back.

**Comment by cb2004** [comment]
Ok, changing the include makes no difference, the form is actually displaying so I am guessing it is correct. In fact when it was an incorrect include I got 3 errors:

Warning: include(./forms/contact.php): failed to open stream: No such file or directory in /var/www/vhosts/SITE/httpdocs/FOLDER/site/templates/views/page.php on line 74

Warning: include(./forms/contact.php): failed to open stream: No such file or directory in /var/www/vhosts/SITE/httpdocs/FOLDER/site/templates/views/page.php on line 74

Warning: include(): Failed opening './forms/contact.php' for inclusion (include_path='.:/usr/share/pear/:/var/www/vhosts/SITE/:/tmp/') in /var/www/vhosts/SITE/httpdocs/FOLDER/site/templates/views/page.php on line 74

Should there be 2 warnings that are the same (top 2)?

**Comment by ryancramerdesign** [comment]
It's definitely not a case matter, I was forgetting that PHP internally keeps all functions as lowercase anyways. One thing I'm seeing in your warnings you pasted in is "templates/views/" whereas I was thinking we were talking about "templates/" before. So that leads to more questions:

We've got a forms/contact.php file and a views/page.php file which includes that contact.php file. So now I'm wondering where the /forms/ directory is located? Since your include statement used "forms/contact.php" or "./forms/contact.php", does that mean that it's located in templates/views/forms/contact.php? I think this is where that include() statement is looking for it. 

If it's instead located in templates/forms/contact.php, then the include() statement in your page.php file would need to be `include("../forms/contact.php");` backtracking by 1 directory. Or you could do `include($config->paths->templates . 'forms/contact.php');` 

Basically, I think there's some confusion from FileCompiler about where the file is actually located, and perhaps PHP is picking it up from the include path. But since FileCompiler isn't able to resolve the path, it's not compiling anything. 

If that's not it, the next thing we should look at is how you are including views/page.php, as maybe that's where the compile/include chain is getting lost.

**Comment by cb2004** [comment]
Ok, in all my includes across my site I have put $config->paths->templates before and it seems to be working. I will just do that going forward to make sure the file compiler is happy.

**Comment by ryancramerdesign** [comment]
An easy rule of thumb with file compiler is that if it sees an include with
a relative path while compiling a file, it can only assume it is relative
to the directory that the file being compiled is in. Using an absolute
path, like you converted to, may reduce any ambiguity when you've got lots
of paths to consider.
